### ISSUE TITLE
[package] Homebrew formula should default to "openjdk" dependency (without a version) for Java projects

### ISSUE THREAD

**Issue by klausbrunner** [jreleaser/jreleaser#1224]
Current default behaviour of JReleaser's brew packager is to add a dependency to "openjdk@VERSION".

This should be changed to "openjdk" without a version:

- Most Java software does not need to be pinned to a specific JDK release, it just has a _minimum_ requirement (e.g. 8 or 11).
- Adding "openjdk" as a dependency will ensure that the _latest_ available OpenJDK release is installed (e.g. 19.0.2 as of this writing).
- Using a specific JDK version for each project may clutter the machine with several JDKs, which is especially useless if those aren't LTS versions.
- In those cases where a specific JDK version must be used, one can still add a specific version dependency (or use a bundled JDK in the first place).

This is also the pattern used by most Homebrew formulae for Java software: 

> Most Java-based packages in Homebrew use the default Homebrew-provided openjdk package for their JDK dependency, although some packages with specific version requirements may use a versioned package such as openjdk@8.
https://docs.brew.sh/Homebrew-and-Java#executing-commands-using-a-different-jdk

**Comment by aalmiray** [comment]
Current workaround is to explicitly disable the default inclusion of `openjdk` dependency and add an explicit dependency, such as

```yaml
    brew:
      active: always
      dependencies:
        openjdk: null
      extraProperties:
        skipJava: true
```

Which generates a formula like the the following one

```ruby
# Generated with JReleaser 1.5.0 at 2023-02-28T09:25:30.019416+01:00
class App < Formula
  desc "Awesome App"
  homepage "https://acme.com/app"
  url "https://github.com/aalmiray/app/releases/download/v1.0.0/app-1.0.0.zip"
  version "1.0.0"
  sha256 "c3b3992bb79c69d22595260bebfa1fc8b567edc6ed43f111d6e22ed645f4378f"
  license "Apache-2.0"

  depends_on "openjdk"

  def install
    libexec.install Dir["*"]
    bin.install_symlink "#{libexec}/bin/app" => "app"
  end

  test do
    output = shell_output("#{bin}/app --version")
    assert_match "1.0.0", output
  end
end
```

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.6.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.6.0))
